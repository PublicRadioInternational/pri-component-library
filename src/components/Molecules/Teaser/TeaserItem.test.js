/**
 * @file TeaserItem.test.js
 * Contains tests for TeaserItem.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import TeaserItem from './TeaserItem.component';

describe('<TeaserItem />', () => {
  it('Matches the Teaser Item Default snapshot', () => {
    const component = renderer
      .create(
        <TeaserItem
          title="Test Title"
          url="/"
          programTitle="test Program"
          programUrl="programs/"
          hasAudio
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Matches the Teaser Item without program snapshot', () => {
    const component = renderer
      .create(<TeaserItem title="Test Title" url="/" hasAudio />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Matches the Teaser Item without program or audio snapshot', () => {
    const component = renderer
      .create(<TeaserItem title="Test Title" url="/" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
