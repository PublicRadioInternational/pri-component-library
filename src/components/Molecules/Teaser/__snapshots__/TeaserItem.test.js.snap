// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeaserItem /> Matches the Teaser Item Default snapshot 1`] = `
<article
  className="teaserItem"
  typeof="sioc:Item foaf:Document"
>
  <div
    className="titleWrap"
  >
    <h4
      className="title"
    >
      <a
        className="link"
        href="/"
      >
        Test Title
      </a>
    </h4>
    <span
      content="Test Title"
      property="dc:title"
    />
  </div>
  <p
    className="program"
  >
    <a
      className="link"
      href="programs/"
    >
      test Program
    </a>
  </p>
</article>
`;

exports[`<TeaserItem /> Matches the Teaser Item without program or audio snapshot 1`] = `
<article
  className="teaserItem"
  typeof="sioc:Item foaf:Document"
>
  <div
    className="titleWrap"
  >
    <h4
      className="title"
    >
      <a
        className="link"
        href="/"
      >
        Test Title
      </a>
    </h4>
    <span
      content="Test Title"
      property="dc:title"
    />
  </div>
</article>
`;

exports[`<TeaserItem /> Matches the Teaser Item without program snapshot 1`] = `
<article
  className="teaserItem"
  typeof="sioc:Item foaf:Document"
>
  <div
    className="titleWrap"
  >
    <h4
      className="title"
    >
      <a
        className="link"
        href="/"
      >
        Test Title
      </a>
    </h4>
    <span
      content="Test Title"
      property="dc:title"
    />
  </div>
</article>
`;
